### PR TITLE
fixes #3592 - obsolete

### PR DESCRIPTION
**Pull request by abenari**

### CODE REVIEW

File: app/views/hosts/show.html.erb [hunk 1]
```diff
@@ -34,12 +33,14 @@
...
-      <div class="tab-pane" id="vm">
-        <%  if @vm %>
-          <% @compute_resource = @host.compute_resource %>
-          <%= render("compute_resources_vms/show/#{@host.compute_resource.provider.downcase}") rescue nil %>
-        <% end %>
-      </div>
+      <%  if @host.compute_resource_id %>
+        <div class="tab-pane" id="vm" data-ajax-url=<%= vm_host_path(@host)%>>
+          <p id="spinner">
+            <%= image_tag '/assets/spinner.gif' %>
```
Comment (ohadlevy) [inline app/views/hosts/show.html.erb]: dont think we need asets?

**Comment by domcleal** [comment]
Looks good, but any chance of keeping the buttons on the top of the page?  Perhaps in a greyed out state if the host is based on a CR, then update the status when the AJAX request returns?

File: app/views/compute_resources_vms/show/_gce.html.erb [hunk 2]
```diff
@@ -1,30 +1,15 @@
...
-    </tr>
-    <tr>
-      <td>Zone</td>
-      <td><%= @vm.zone %></td>
-    </tr>
-    <tr>
-      <td>Private IP</td>
-      <td><%= @vm.private_ip_address %></td>
-    </tr>
+    <tr><th colspan="2">Properties</th></tr>
```
Comment (domcleal) [inline app/views/compute_resources_vms/show/_gce.html.erb]: _() here

**Comment by ekohl** [comment]
I can confirm it works using the vmware provider, but I agree with @domcleal that keeping the power options + console in the UI would be beneficial.

**Comment by ohadlevy** [comment]
@domcleal would you prefer to break this down, or review as is?

**Comment by domcleal** [comment]
@ohadlevy I'm not sure... @abenari what's the state, what's changed?

When I reviewed it when I last commented, it all looked pretty good to me, I don't recall having any other comments to make.  I can see it's been updated, but what's changed?

**Comment by abenari** [comment]
@domcleal this patch now does a few things:
1. lazy load vm on host show page, and place the power buttons on the action bar.
2. add multiple power actions to vm (stop, power off, reset, reboot instead of just stop)
3. uses the PowerManager class for vm power action.
4. reorganize helpers and view code for power actions that was scattered in multiple places .

File: app/helpers/compute_resources_vms_helper.rb [hunk 3]
```diff
@@ -1,10 +1,30 @@
 module ComputeResourcesVmsHelper
 
+  def vm_power_actions(vm)
+    power_manager = PowerManager::Virt.new(:vm=>vm)
+    power_manager.supported_actions.each_with_index.map do |action, i|
+      action_name = action.to_s.humanize
```
Comment (domcleal) [inline app/helpers/compute_resources_vms_helper.rb:6]: This needs to be a bit more careful for i18n, as we're extracting lower case strings in POWER_ON/OFF_ACTIONS so we get "reboot", "poweroff" etc translated.  This code tries to translate the humanised versions of "Reboot" etc, which aren't translated (they're case sensitive).  Another minor issue is that it should be "power off" in English.

I'd suggest changing the power manager constants to hashes of method names to user-friendly strings and iterate over key/value pairs.

File: app/helpers/bmc_helper.rb [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
     end
   end
 
-  def power_actions
-    action_buttons(
-      (PowerManager::SUPPORTED_ACTIONS - ['state']).map do |action|
+  def power_actions host
+    toolbar_action_buttons(
+      host.power.supported_actions.map do |action|
         display_link_if_authorized(_(action.to_s.capitalize), { :action => "power", :id => @host, :power_action => action},
```
Comment (domcleal) [inline app/helpers/bmc_helper.rb:14]: ditto here regarding i18n

**Comment by domcleal** [comment]
Nice work, that's much better.  One comment above about some i18n problems.  Sorry for the delay reviewing!

**Comment by domcleal** [comment]
ping @abenari?

**Comment by angystardust** [comment]
Any hope to see this one in 1.5? I think it's pretty useful for people with a lot of vm managed by a CR.

**Comment by abenari** [comment]
code is obsolete, this pr was replaced with a new one.  #1349

**Pull request closed without merging**